conjure-core/src/main/java/com/palantir/conjure/defs/validator/ErrorDefinitionValidator.java

```diff
@@ -33,5 +36,16 @@ public static void validate(ErrorDefinition definition) {
                 Stream.concat(definition.getSafeArgs().stream(), definition.getUnsafeArgs().stream())
                         .map(FieldDefinition::getFieldName)
                         .collect(Collectors.toSet()));
+        ImmutableList<FieldName> fieldsDeclaringSafety = Stream.concat(
+                        definition.getSafeArgs().stream(), definition.getUnsafeArgs().stream())
+                .filter(arg -> arg.getSafety().isPresent())
+                .map(FieldDefinition::getFieldName)
+                .collect(ImmutableList.toImmutableList());
+        if (!fieldsDeclaringSafety.isEmpty()) {
+            throw new ConjureIllegalStateException("ErrorDefinition field safety is defined by the key 'safeArgs' or "
+                    + "'unsafeArgs', safety cannot be declared: "
+                    + fieldsDeclaringSafety);
+        }
+        // TODO(ckozak): Resolve safety of all referenced types to ensure alignment with the encapsulating field.
```

Comment:
Unsure if we want to do this, if so, likely not here. This project doesn't currently need to be aware of how safety is evaluated, only where it's allowed. That would need to change for us to validate at this point -- I can add annotations to the generated errors such that known-unsafe data cannot be passed in, which should be sufficient.